DOCS-877: Update single-axis gantry, clarify limit_pins (#1429)
viamrobotics/docs: docs/components/gantry/single-axis.md

| --------- | ---- | --------- | ----------  |
| `length_mm` | int | **Required** | Length of the axis of the gantry in millimeters. |
| `motor` | string | **Required** | `name` of the [motor](/components/motor/) that moves the gantry's actuator. |
| `board`  |  string | Optional | `name` of the [board](/components/board/) containing the limit switches and pins. If `limit_pins` exist, `board` is required. |

> | `limit_pins`  | object | Optional | The pins on the `board` which are attached to the limit switches on either end. If the [motor](/components/motor/) used does not include an [encoder](/components/motor/gpio/encoded-motor/), you must set `limit_pins`.The order of these pins is important, as the gantry may try to travel beyond the limit switch (or switches) if the pins are specified in the wrong order. The switch representing the home (`0`) position must be first in the list. In addition, if you are configuring multiple `single-axis` gantries with `limit_pins`, you can [specify the order of homing routines](#specify-the-order-of-homing-routines) if needed. |

Reviewer: same for this comment about specifying the order

| `limit_pin_enabled_high` | boolean | Optional | Whether the limit pin must be “high” or “low” to be considered “hit”, with a value of `true` representing "high". This attribute is **required** if `limit_pins` is set.<br> Default: `false` |
| `mm_per_rev` | int | **Required** | How far the gantry moves (linear, distance in mm) per one revolution of the motor’s output shaft. This typically corresponds to Distance = PulleyDiameter * pi, or the pitch of a linear screw. |
| `gantry_mm_per_sec` | int | Optional | The speed at which the gantry moves in millimeters per second. Used to calculate the gantry `motor`'s revolutions per minute (RPM). <br> Default: `100` RPM |

## Specify the order of homing routines

> When `limit_pins` is set, a `single-axis` gantry component performs a homing routine when `viam-server` starts.

Reviewer: no longer true

> The axis moves towards one limit switch, then the other, before stopping at 80% of the travel of the axis.

Reviewer: now 50%

If you configure multiple `single-axis` gantries (for example, when using a [`multi-axis` gantry](/components/gantry/multi-axis/)), the order in which these homing routines are executed may be important.
For example, if there is a tool mounted to the z axis gantry that could be damaged during homing of the x or y axes gantries, you would want to ensure that the z axis gantry completes its homing routine first, so that the tool head returns to a safe location before the homing process begins on the other axes.

> In this case, you can specify the axes that must be homed first in a `depends_on` array to each `single-axis` gantry.

Reviewer: just want to make sure nothing is missed, this is no longer necessary, order can be specified in multi-axis array

The example below configures `myFirstGantry` to wait until the homing process for `mySecondGantry` has completed before starting its own:

```json {class="line-numbers linkable-line-numbers"}
{
    "components": [
        // < Your motor config >
        {
            "name": "myFirstGantry",
            "type": "gantry",
            "model": "single-axis",
            "attributes": {
                "length_mm": 1000,
                "board": "local",
                "limit_pin_enabled_high": false,
                "limit_pins": [
                    "32",
                    "36"
                ],
                "depends_on": [
                    "mySecondGantry"
                ]
            }
        },
        // < mySecondGantry config >
    ]
}
```

{{% alert="Note" color="note" %}}

> The order of axes in the `subaxes_list` for a [`multi-axis` gantry](/components/gantry/multi-axis/) does not influence the order of homing.

Reviewer: now the order is the order they will home in if `Home` is called on the multi-axis gantry

For example, if there is a tool mounted to the z axis gantry that could be damaged during homing of the x or y axes gantries, you would want to ensure that the z axis gantry completes its homing routine first, so that the tool head returns to a safe location before the homing process begins on the other axes.

In this case, you can specify the axes that must be homed first in a `depends_on` array to each `single-axis` gantry.
The example below configures `myFirstGantry` to wait until the homing process for `mySecondGantry` has completed before starting its own:

```json {class="line-numbers linkable-line-numbers"}
{
    "components": [
        // < Your motor config >
        {
            "name": "myFirstGantry",
            "type": "gantry",
            "model": "single-axis",
            "attributes": {
                "length_mm": 1000,
                "board": "local",
                "limit_pin_enabled_high": false,
                "limit_pins": [
                    "32",
                    "36"
                ],
                "depends_on": [
                    "mySecondGantry"
                ]
            }
        },
        // < mySecondGantry config >
    ]
}
```

> {{% alert="Note" color="note" %}}

Reviewer 2: ```suggestion
{{% alert title="Note" color="note" %}}
```

| `limit_pin_enabled_high` | boolean | Optional | Whether the limit pin must be “high” or “low” to be considered “hit”, with a value of `true` representing "high". This attribute is **required** if `limit_pins` is set.<br> Default: `false` |
| `mm_per_rev` | int | Optional | How far the gantry moves (linear, distance in mm) per one revolution of the motor’s output shaft. This typically corresponds to Distance = PulleyDiameter * pi, or the pitch of a linear screw. |
| `gantry_rpm` | int | Optional | The gantry `motor`'s default revolutions per minute (RPM). |

## Specify the order of homing routines

When `limit_pins` is set, a `single-axis` gantry component performs a homing routine when `viam-server` starts.

> The axis will move towards one limit switch, then the other, before stopping at 80% of the travel of the axis.

Reviewer 2: ```suggestion
The axis moves towards one limit switch, then the other, before stopping at 80% of the travel of the axis.
```
[opt/nit] We often try to put things in present tense

> If there are multiple `single-axis` gantries configured (for example, when using a [`multi-axis` gantry](/components/gantry/multi-axis/)), the order in which these homing routines are executed may be important.

Reviewer 2: ```suggestion
If you configure multiple `single-axis` gantries (for example, when using a [`multi-axis` gantry](/components/gantry/multi-axis/)), the order in which these homing routines are executed may be important.
```
[nit] speak to user. Also, when it says "...may be important," can we be more clear about whether it is or isn't?

Reviewer: maybe something like "If you configure multiple `single-axis` gantries into a `multi-axis` gantry, the axes will home in the order they are listed in the `multi-axis` gantry"

In this case, you can specify the axes that must be homed first in a `depends_on` array to each `single-axis` gantry.

> For example, if there is a tool mounted to the z axis that could be damaged during homing of the x or y axes, you should add the z axis as a dependency to the x and y axes in your configuration. This will ensure the z axis completes its homing routine first, and that the tool head returns to a safe location to begin the homing process on the other axes.

Reviewer 2: ```suggestion
For example, if there is a tool mounted to the z axis that could be damaged during homing of the x or y axes, add the z axis as a dependency to the x and y axes in your configuration.
This dependency ensures that the z axis completes its homing routine first, and that the tool head returns to a safe location before the homing process begins on the other axes.
```
main nit: newline. Additional nits are opt.

Reviewer: This actually may not be necessary info anymore as the homing sequences now need to be called individually, they no longer run automatically on building

> If there are multiple `single-axis` gantries configured (for example, when using a [`multi-axis` gantry](/components/gantry/multi-axis/)), the order in which these homing routines are executed may be important.
> 
> In this case, you can specify the axes that must be homed first in a `depends_on` array to each `single-axis` gantry.
> For example, if there is a tool mounted to the z axis that could be damaged during homing of the x or y axes, you should add the z axis as a dependency to the x and y axes in your configuration. This will ensure the z axis completes its homing routine first, and that the tool head returns to a safe location to begin the homing process on the other axes.

Reviewer 2: Realized changing the order might make more sense:
```suggestion
If you configure multiple `single-axis` gantries (for example, when using a [`multi-axis` gantry](/components/gantry/multi-axis/)), the order in which these homing routines are executed may be important.
For example, if there is a tool mounted to the z axis that could be damaged during homing of the x or y axes, you want the z axis to complete its homing routine first.

In this case, you can specify the axes that must be homed first in a `depends_on` array to each `single-axis` gantry.
Add the z axis as a dependency to the x and y axes in your configuration to ensure that the tool head returns to a safe location before the homing process begins on the other axes.
```

{{% alert="Note" color="note" %}}

> The order of axes in the `subaxes_list` for a `multi-axis` gantry does not influence the order of homing.
> A `multi-axis` gantry component is not constructed until all the `single-axis` gantry components are constructed and finish homing.

Reviewer 2: ```suggestion
A `multi-axis` gantry component is not constructed until all the `single-axis` gantry components are constructed and finish homing.
However, the order of axes in the `subaxes_list` for a `multi-axis` gantry does not influence the order of homing.
```
I think this makes a little more sense? I think the existing second sentence is a slightly separate concept from the existing first and last sentences, so maybe it can actually come out of this note box entirely? Or consider reordering thusly.

Author: I see what you're saying! However, I think removing the second sentence entirely is the play here: it's an implementation detail that the user won't really interact with (how we construct mult-axis components on our end). From the user perspective: they just need to know that homing order is controllable only with `depends_on`, and not by any ordering of the `subaxes_list` to `multi-axis` (which, conceptually, could otherwise seem a logical assumption). Pulling that out seems to make this admonition much more focused and clearer -- LMK if you agree!

Reviewer 2: Yes, I do agree! That part looks good
